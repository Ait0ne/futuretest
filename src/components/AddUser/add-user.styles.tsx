import styled from 'styled-components';

export const AddUserContainer = styled.div`
display: flex;
flex-direction: column;
justify-content: center;
align-items: flex-start;
margin:50px 0 20px 0;
`