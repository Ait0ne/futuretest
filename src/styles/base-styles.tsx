import styled from 'styled-components';

interface ButtonProps {
    active?: boolean
}

export const Button = styled.button<ButtonProps>`
cursor: pointer;
min-width: 64px;
padding: 6px 16px;
color: rgba(0, 0, 0, 0.87);
text-transform: uppercase;
font-weight: 500;
border-radius:4px;
border: none;
margin: 5px;
background-color: #e0e0e0;
box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2), 0px 2px 2px 0px rgba(0,0,0,0.14), 0px 1px 5px 0px rgba(0,0,0,0.12);
&:disabled {
    color: rgba(0, 0, 0, 0.26)
}
`

export const Input = styled.input`
padding: 12px 25px 12px 14px;
border-radius: 4px;
width:25ch;
border: 1px solid rgba(0,0,0,0.3);
font-size: 1rem;
margin: 5px 10px;
&:focus {
    outline:none;
}

`