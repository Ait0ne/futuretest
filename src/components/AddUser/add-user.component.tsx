import React, { useState } from 'react';

import {Button} from '../../styles/base-styles';
import {AddUserContainer} from './add-user.styles';
import {IUser} from '../UsersListProvider/users-list-provide.component';
import AddUserForm from '../AddUserForm/add-user-form.component';

export interface AddUserProps {
    addRow: (user:IUser) => void
}



const AddUser:React.FC<AddUserProps> = ({addRow}) => {
    const [showAddUserForm, setShowAddUserForm] = useState(false)

    const toggleAddUserForm = () => {
        setShowAddUserForm(c=> !c)
    }

    return (
        <AddUserContainer>
            <Button onClick={toggleAddUserForm}>Добавить</Button>
            {
                showAddUserForm?<AddUserForm addRow={addRow}/>:null
            }
        </AddUserContainer>
    )
}

export default AddUser;