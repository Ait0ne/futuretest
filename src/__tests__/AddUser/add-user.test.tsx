import React from 'react';
import {shallow} from 'enzyme';

import AddUser from '../../components/AddUser/add-user.component';
import {Button} from '../../styles/base-styles';

describe('AddUser', ()=> {
    const addUser = shallow(<AddUser addRow={jest.fn()}/>)

    it('renders correctly', () => {
        expect(addUser).toMatchSnapshot()
    })

    it('renders a button', () => {
        expect(addUser.find(Button).length).toEqual(1)
    })

    it('doesnt render AddUserForm initially', () => {
       expect(addUser.find('AddUserForm').length).toEqual(0) 
    })

    it('shows form on button click', () => {
        const button = addUser.find(Button)
        button.simulate('click')
        expect(addUser.find('AddUserForm').length).toEqual(1) 
    })

})