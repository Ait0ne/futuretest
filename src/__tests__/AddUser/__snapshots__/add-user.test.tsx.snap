// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddUser renders correctly 1`] = `
<styled.div>
  <styled.button
    onClick={[Function]}
  >
    Добавить
  </styled.button>
</styled.div>
`;
